### ISSUE TITLE
systemd-udevd makefile target should depend on generated headers

### ISSUE THREAD

**Issue by grossws** [systemd/systemd#5727]
### Submission type

  - [x] Bug report
  - [ ] Request for enhancement (RFE)

### systemd version the issue has been seen with

> systemd-233

### Used distribution

> ArchLinux, but irrelevant in this case

### In case of bug report: Expected behaviour you didn't see

> `make systemd-udevd` should build `systemd-udevd`

### In case of bug report: Unexpected behaviour you saw

> `make systemd-udev` fails with 
```
src/udev/udev-builtin-keyboard.c:33:37: fatal error: keyboard-keys-from-name.h: No such file or directory
 #include "keyboard-keys-from-name.h"
```

Currently it requires something like 
```bash
make src/basic/{arphrd,af,cap,errno}-{from,to}-name.h src/journal/audit_type-{from,to}-name.h src/udev/keyboard-keys-from-name.h

make systemd-udevd
```

### In case of bug report: Steps to reproduce the problem

```bash
tar xzf v233.tar.gz
cd systemd-233/

./autogen.sh
./configure

make systemd-udevd
```

### Comments

Maybe some intermediate target for generated headers would be appropriate. Or explicit dependency on generated headers.

**Comment by evverx** [comment]
Have you tried
```
make built-sources systemd-udevd
```
?
See 7ee7b225bd2fc3e.

**Comment by fsateler** [comment]
Duplicate of #2841 and #503

**Comment by grossws** [comment]
@evverx, @fsateler thanks.

I guess, it would be a good idea to add info about `built-sources` target to `HACKING` or `README`. Even if it's known and intended behavior of autotools its quite surprising for somebody not so familiar with autoconf/automake.

**Comment by fsateler** [comment]
@grossws Lets hope we can leave autofoo behind after #5704 ... Meson does not have that problem
